File: tests/unit/data/full/Virtual/PYLIB/board0/rioclient.c

#include <unistd.h>
#include <stdint.h>
#include <stdlib.h>
#include <stdbool.h>
#include <stdio.h>
#include <string.h>

#include <rio.h>

data_t *data;

int set_values(void) {
    *data->SIGOUT_BOARD0_BOARD0_WLED_0_GREEN = 0;
    *data->SIGOUT_BOARD0_BOARD0_WLED_0_BLUE = 0;
    *data->SIGOUT_BOARD0_BOARD0_WLED_0_RED = 0;
    *data->SIGOUT_BOARD0_STEPDIR0_VELOCITY = 0.0;
    *data->SIGOUT_BOARD0_STEPDIR0_ENABLE = 0;
    *data->SIGOUT_BOARD0_STEPDIR1_VELOCITY = 0.0;
    *data->SIGOUT_BOARD0_STEPDIR1_ENABLE = 0;
    *data->SIGOUT_BOARD0_STEPDIR2_VELOCITY = 0.0;
    *data->SIGOUT_BOARD0_STEPDIR2_ENABLE = 0;
}

int print_values(void) {
    printf("SIGIN_BOARD0_STEPDIR0_POSITION: %f\n", *data->SIGIN_BOARD0_STEPDIR0_POSITION);
    printf("SIGIN_BOARD0_STEPDIR1_POSITION: %f\n", *data->SIGIN_BOARD0_STEPDIR1_POSITION);
    printf("SIGIN_BOARD0_STEPDIR2_POSITION: %f\n", *data->SIGIN_BOARD0_STEPDIR2_POSITION);
    printf("\n");
}

int main(int argc, char **argv) {
    data = init(argc, argv);

    while (1) {
        set_values();
        rio_readwrite(NULL, 0);
        print_values();

        usleep(100000);
    }

    return 0;
}

File: tests/unit/data/full/satmcu/PYLIB/board0/rioclient.py
#!/usr/bin/env python3
#
#

import sys
import time
from rio import RioWrapper

def set_values(rio):
    rio.data_set("SIGOUT_BOARD0_STEPDIR0_VELOCITY", 0.0)
    rio.data_set("SIGOUT_BOARD0_STEPDIR0_ENABLE", 0)
    rio.data_set("SIGOUT_BOARD0_STEPDIR1_VELOCITY", 0.0)
    rio.data_set("SIGOUT_BOARD0_STEPDIR1_ENABLE", 0)
    rio.data_set("SIGOUT_BOARD0_STEPDIR2_VELOCITY", 0.0)
    rio.data_set("SIGOUT_BOARD0_STEPDIR2_ENABLE", 0)
    rio.data_set("SIGOUT_BOARD0_BITOUT0_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_BITOUT1_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT0_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT1_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT2_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT3_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT4_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT5_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT6_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT7_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU0_GPIOOUT8_BIT", 0)
    rio.data_set("SIGOUT_BOARD0_SATMCU1_GPIOOUT9_BIT", 0)

def print_values(rio):
    for name, config in rio.data_info().items():
        if config["direction"] == "input":
            print(f'{config["halname"]} = {rio.data_get(name)}')
    print("")

rio = RioWrapper(sys.argv)

while True:
    set_values(rio)
    rio.rio_readwrite()
    print_values(rio)
    time.sleep(0.1)
